src/opm/output/eclipse/DoubHEAD.cpp

```diff
@@ -0,0 +1,603 @@
+    dh_166  = 166,
+    dh_167  = 167,
+    dh_168  = 168,
+    dh_169  = 169,
+
+    // 170..179
+    dh_170  = 170,
+    dh_171  = 171,
+    dh_172  = 172,
+    dh_173  = 173,
+    dh_174  = 174,
+    dh_175  = 175,
+    dh_176  = 176,
+    dh_177  = 177,
+    dh_178  = 178,
+    dh_179  = 179,
+
+    // 180..189
+    dh_180  = 180,
+    dh_181  = 181,
+    dh_182  = 182,
+    dh_183  = 183,
+    dh_184  = 184,
+    dh_185  = 185,
+    dh_186  = 186,
+    dh_187  = 187,
+    dh_188  = 188,
+    dh_189  = 189,
+
+    // 190..199
+    dh_190  = 190,
+    dh_191  = 191,
+    dh_192  = 192,
+    dh_193  = 193,
+    dh_194  = 194,
+    dh_195  = 195,
+    dh_196  = 196,
+    dh_197  = 197,
+    dh_198  = 198,
+    dh_199  = 199,
+
+    // 200..209
+    dh_200  = 200,
+    dh_201  = 201,
+    dh_202  = 202,
+    dh_203  = 203,
+    dh_204  = 204,
+    dh_205  = 205,
+    dh_206  = 206,
+    dh_207  = 207,
+    dh_208  = 208,
+    dh_209  = 209,
+
+    // 210..219
+    dh_210  = 210,
+    dh_211  = 211,
+    dh_212  = 212,
+    dh_213  = 213,
+    dh_214  = 214,
+    dh_215  = 215,
+    dh_216  = 216,
+    dh_217  = 217,
+    dh_218  = 218,
+    dh_219  = 219,
+
+    // 220..227
+    dh_220  = 220,
+    dh_221  = 221,
+    dh_222  = 222,
+    dh_223  = 223,
+    dh_224  = 224,
+    dh_225  = 225,
+    dh_226  = 226,
+    dh_227  = 227,
+    dh_228  = 228,
+
+// ---------------------------------------------------------------------
+// ---------------------------------------------------------------------
+
+    NUMBER_OF_ITEMS        // MUST be last element of enum.
+};
+
+namespace {
+    /// Convert std::tm{} to Date-Number.
+    ///
+    /// ECL Restrictions:
+    ///   - Calendar start: 1st January 0000
+    ///   - Year length: 365.25 days
+    ///   - No special leap year handling
+    ///
+    /// \param[in] year tm::tm_year of date (years since 1900).
+    ///
+    /// \param[in] yday tm::tm_yday of date (days since 1st January).
+    ///
+    /// \return Date-number corresponding to specified day-of-year in
+    ///    specified year, subject to restrictions outlined above.
+    double toDateNum(const int year, const int yday)
+    {
+        return std::floor(365.25 * (year + 1900))
+            + (yday + 1); // Day of year [1 .. 365]
+    }
+
+    double toDateNum(const std::chrono::time_point<std::chrono::system_clock> tp)
+    {
+        const auto t0  =  std::chrono::system_clock::to_time_t(tp);
+        const auto tm0 = *std::gmtime(&t0);
+
+        // Set clock to 01:00:00+0000 on 1901-<month>-<day> to get
+        // "accurate" day-of-year calculation (no leap year, no DST offset,
+        // not previous day reported).
+        auto tm1 = std::tm{};
+        tm1.tm_year = 1;        // 1901
+        tm1.tm_mon  = tm0.tm_mon;
+        tm1.tm_mday = tm0.tm_mday;
+
+        tm1.tm_hour  = 1;
+        tm1.tm_min   = 0;
+        tm1.tm_sec   = 0;
+        tm1.tm_isdst = 0;
+
+        const auto t1 = ::Opm::RestartIO::makeUTCTime(tm1);
```

Comment:
After quite a bit of reading docs and debugging, I have at last found a bug here: makeUTCTime() cannot be called with time points before the epoch, since then the mktime() call inside will return -1. That is not accounted for. The test failure I get is:
```
108: /Users/atgeirr/opm/opm-common/tests/test_DoubHEAD.cpp:88: error: in "Member_Functions/Time_Stamp": difference{0.000133156} between v[161 - 1]{735979} and 736077.0{736077} exceeds 1e-10%
108: /Users/atgeirr/opm/opm-common/tests/test_DoubHEAD.cpp:91: error: in "Member_Functions/Time_Stamp": difference{0.000133134} between v[162 - 1]{736102} and 736200.0{736200} exceeds 1e-10%
108: 
108: *** 2 failures are detected in the test module "DoubHEAD_Vector"
```
(note that line numbers have changed a little from this version).

Reply:
Another issue: I am not at all sure if the ECL concept accounts for time zones or not. When the deck says:
```
START
9 APR 2015 /
```
I think we should assume no DST for example. The time point of the test (which not coincidentally is the date above) is given as a std::tm with tm_isdst = 0, but on that date in my timezone DST was active, so I get a UTC time point of April 8, 2300, which probably is why I get 98 days off instead of 99 days (which is what I would expect from the bug described above).

TL;DR: we should probably be totally timezone-agnostic in all C library usage. We should assume that we operate on UNIX epoch (which is guaranteed for system_clock from C++20 but in practice I think it is true for all systems), and never involve dates before that. The offset from year 0 to 1970 (in "ECL calendar") is a fixed number of days that we could store.